tests/unit_tests/base32.cpp

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2023, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "gtest/gtest.h"
+
+#include <cstddef>
+#include <cstdlib>
+#include <fstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "common/base32.h"
+#include "crypto/crypto.h"
+#include "string_tools.h"
+#include "unit_tests_utils.h"
+
+static size_t num_prefix_similar(const char* a, const char* b, size_t n)
+{
+    size_t i;
+    for (i = 0; i < n && a[i] == b[i]; ++i) {}
+    return i;
+}
+
+static constexpr size_t estimate_b32_encoding_size(const size_t l)
+{
+    return static_cast<size_t>(std::ceil(l * 8.0f / 5.0f));
+}
+
+static std::string hex_decode(const std::string& s)
+{
+    std::string res;
+    if (!epee::string_tools::parse_hexstr_to_binbuff(s, res))
+        throw std::runtime_error("hex decode");
+    return res;
+}
+
+TEST(base32, encode_decode)
+{
+    for (size_t raw_len = 0; raw_len < 250; ++raw_len)
+    {
+        for (size_t i = 0; i < 10; ++i)
+        {
+            std::string raw_buf;
+            raw_buf.resize(raw_len);
+            crypto::generate_random_bytes_not_thread_safe(raw_buf.size(), &raw_buf[0]);
+
+            const std::string encoded_buf = base32::encode(raw_buf);
+            const std::string decoded_buf = base32::decode(encoded_buf);
+
+            ASSERT_EQ(raw_buf, decoded_buf);
+        }
+    }
+}
+
+TEST(base32, jamtis_address_prefix_compat)
```

Comment:
A short comment explaining what this asserts?

I had to look up the Jamtis spec to (hopefully correctly) understand that the address prefix is base32.

Reply:
That the Jamtis address header characters can be decoded (given a correct length).